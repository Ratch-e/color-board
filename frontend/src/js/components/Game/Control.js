import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Control = ({color, click}) => (
	<div
		className={classNames('controls__item', `controls__item_color_${color}`)}
		onClick={() => click(color)}
	/>
);

Control.propTypes = {
	color: PropTypes.string,
	click: PropTypes.func,
};

export default Control;