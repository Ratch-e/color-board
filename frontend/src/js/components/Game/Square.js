import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Square = ({color, click, x, y}) => (
	<div
		className={classNames('board__square', `board__square_color_${color}`)}
		onClick={() => click(x, y, color)}
	/>
);

Square.propTypes = {
	color: PropTypes.string,
	click: PropTypes.func,
	x: PropTypes.number,
	y: PropTypes.number,
};

export default Square;