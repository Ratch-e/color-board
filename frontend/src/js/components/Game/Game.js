import React, {Component} from 'react';
import Square from './Square';

class Game extends Component {
	constructor(props) {
		super(props);

		this.state = {
			colors: [
				[],[],[],[],[],[],[],[]
			],
		};
		this.setColors = this.setColors.bind(this);
		this.squareClick = this.squareClick.bind(this);
		this.createBoard = this.createBoard.bind(this);
	}

	componentDidMount() {
		this.setColors();
	}

	/**
	 * Установка случайных цветов при старте
	 */
	setColors() {
		const COLORS = [
			'red', 'green', 'blue'
		];
		const row = 8, col = 8;
		let newState = {...this.state};

		for (let x = 0; x < row; x++) {
			newState.colors[x] = [];
			for (let y = 0; y < col; y++) {
				newState.colors[x][y] = COLORS[Math.floor(Math.random() * COLORS.length)];
			}
		}

		this.setState({colors: newState.colors});
	}

	/**
	 * Генератор клеток
	 * @returns {Array}
	 */
	createBoard() {
		let squares = [];
		const row = 8, col = 8;
		for (let x = 0; x < row; x++) {
			for (let y = 0; y < col; y++) {
				squares.push(<Square key={squares.length} color={this.state.colors[x][y]} click={this.squareClick} x={x} y={y}/>);
			}
		}
		return squares;
	}

	squareClick(x, y, color) {
		let newState = {...this.state};
		newState.colors[0][0] = color;
		this.setState({colors: newState.colors});
		return null;
	}

	render() {
		return (
			<div className="board">
				{this.createBoard()}
			</div>
		);
	}
}

export default Game;
