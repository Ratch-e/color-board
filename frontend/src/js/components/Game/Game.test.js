import React from 'react';
import { shallow } from 'enzyme';
import Game from './Game';

describe('Основное окно игры', () => {
	it('рендерится корректно', () => {
		const component = shallow(<Game />);
		expect(component.length).toMatchSnapshot();
	});
});
