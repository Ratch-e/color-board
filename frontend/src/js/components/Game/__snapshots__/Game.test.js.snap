// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Основное окно игры рендерится корректно 1`] = `1`;
